docs/bos/queryapi/how-works.md

```diff
@@ -0,0 +1,45 @@
+---
+id: how-it-works
+title: How QueryAPI works
+sidebar_label: How it works
+---
+
+QueryApi is a streaming indexer implementation that executes queries written by developers on the NEAR blockchain.
+QueryApi allows hosted execution of complex queries (ones that can’t be answered by a [simple RPC](../../5.api/rpc/introduction.md) or [Enhanced API](https://docs.pagoda.co/api) call) and hosting of the data output of those queries using GraphQL endpoints.
+
+
+## Components involved 
+
+The QueryApi implementation integrates many different components in a single and streamlined solution.
+In a high-level overview, the main components are:
+
+:::info Components
+`NEAR Protocol` -> `NEAR Lake` -> `Coordinator` -> `Database` -> `API`
```

Comment:
```suggestion
`NEAR Protocol` -> `NEAR Lake` -> `Coordinator` -> `Runner` -> `Database` -> `API`
```
We might want to mention that `Runner` executes the user code which outputs the data to the database?